components/cpu/component.go

```diff
@@ -167,3 +155,352 @@ func (c *component) RegisterCollectors(reg *prometheus.Registry, dbRW *sql.DB, d
 	c.gatherer = reg
 	return metrics.Register(reg, dbRW, dbRO, tableName)
 }
+
+var (
+	prevMu sync.RWMutex
+	prev   *cpu.TimesStat
+)
+
+func setPrevTimeStat(t cpu.TimesStat) {
+	prevMu.Lock()
+	defer prevMu.Unlock()
+	prev = &t
+}
+
+func getPrevTimeStat() *cpu.TimesStat {
+	prevMu.RLock()
+	defer prevMu.RUnlock()
+	return prev
+}
+
+// CheckOnce checks the current pods
+// run this periodically
+func (c *component) CheckOnce() {
+	log.Logger.Infow("checking cpu")
+	d := Data{
+		ts: time.Now().UTC(),
+	}
+	metrics.SetLastUpdateUnixSeconds(float64(d.ts.Unix()))
+	defer func() {
+		c.lastMu.Lock()
+		c.lastData = &d
+		c.lastMu.Unlock()
+	}()
+
+	d.Info = &Info{}
+	arch, err := host.KernelArch()
+	if err != nil {
+		d.Info.err = err
+		return
+	}
+
+	cctx, ccancel := context.WithTimeout(c.ctx, 10*time.Second)
+	infos, err := cpu.InfoWithContext(cctx)
+	ccancel()
+	if err != nil {
+		d.Info.err = err
+		return
+	}
+	if len(infos) > 0 {
+		d.Info.Arch = arch
+		d.Info.CPU = infos[0].ModelName
+		d.Info.Family = infos[0].Family
+		d.Info.Model = infos[0].Model
+		d.Info.ModelName = infos[0].ModelName
+	}
+
+	d.Cores = &Cores{}
+	cctx, ccancel = context.WithTimeout(c.ctx, 10*time.Second)
+	physicalCores, err := cpu.CountsWithContext(cctx, false)
+	ccancel()
+	if err != nil {
+		d.Cores.err = err
+		return
+	}
+	cctx, ccancel = context.WithTimeout(c.ctx, 10*time.Second)
+	logicalCores, err := cpu.CountsWithContext(cctx, true)
+	ccancel()
+	if err != nil {
+		d.Cores.err = err
+		return
+	}
+	d.Cores.Physical = physicalCores
```

Comment:
Perhaps we can use https://pkg.go.dev/github.com/prometheus/procfs instead of these... it's more simple and used by node-exporter..